t/run/runenv_hashseed.t

```diff
@@ -16,121 +31,193 @@ skip_all_without_config('d_fork');
 skip_all("NO_PERL_HASH_ENV or NO_PERL_HASH_SEED_DEBUG set")
     if $Config{ccflags} =~ /-DNO_PERL_HASH_ENV\b/
     || $Config{ccflags} =~ /-DNO_PERL_HASH_SEED_DEBUG\b/;
-
-my @modes = (
-        'NO',
-        'RANDOM',
-        'DETERMINISTIC'
-    ); # 0, 1 and 2 respectively
-my $repeat = 50;
-plan tests => @modes * 7 * $repeat;
 use strict;
-our $TODO;
+use warnings;
 
+# enable DEBUG_RUNENV if you want to see what is being returned
+# by the executed perl.
 sub my_runperl_and_capture {
-    my ($opts_hash, $cmd_array)= @_;
+    my ($opts_hash, $cmd_array) = @_;
     my ( $out, $err )
         = runperl_and_capture( $opts_hash, $cmd_array );
     my $err_got_data = "";
     while ($err=~s/^(Got.*\n)//) {
         $err_got_data .= $1;
     }
-    my @rand_bits_data;
+    my @rand_bits;
     while ($err=~s/^(PL_hash_rand_bits=.*)\n//m) {
-        push @rand_bits_data, $1;
+        push @rand_bits, $1;
     }
-    return ($out, $err, $err_got_data, \@rand_bits_data);
+    $err =~ /HASH_SEED = (0x[a-f0-9]+)/
+        or die "Failed to extract hash seed from runperl_and_capture";
+    my $seed = $1;
+    return ($out, $err, $seed, $err_got_data, \@rand_bits);
 }
 
-# Test that PERL_PERTURB_KEYS works as expected.  We check that we get the same
-# results if we use PERL_PERTURB_KEYS = 0 or 2 and we reuse the seed from previous run.
+my @modes = (
+        'NO',
+        'RANDOM',
+        'DETERMINISTIC'
+    ); # 0, 1 and 2 respectively
+
+my $repeat = 50; # if this changes adjust the comments below.
+my $min_buckets = 100_000;
+my $actual_buckets = 1;
+$actual_buckets *= 2 while $actual_buckets <= $min_buckets;
+my $key_expr = '"A" .. "Z", "a" .. "z", 0 .. 9';
+my @keys = eval $key_expr;
+my $allowed_fails = 2; # FIXME: compute from @keys and $actual_buckets
```

Comment:
Until then, maybe add a comment that we expect this to give a false negative about 1 in 1.1e11 times. You can add "Hugo says" on there if you like. :)

Reply:
Sure, ill add something like that. I am not sure exactly what the odds are, given that some changes to the mask might not actually change the order. Eg, if nothing is in the first bucket then changes to only the low bit of the mask wont change the order of the keys, for instance if the mask was 0x0 and 0x1. That was part of the reason I added more keys. I think stating a specific probability is not helpful unless we know we are right.  That was part of the reason I changed some of the other wording.